typescript/cli/src/tests/warp-apply.e2e-test.ts

```diff
@@ -0,0 +1,132 @@
+import { expect } from 'chai';
+import { Wallet } from 'ethers';
+import { beforeEach } from 'mocha';
+
+import { ChainAddresses } from '@hyperlane-xyz/registry';
+import { TokenRouterConfig, TokenType } from '@hyperlane-xyz/sdk';
+
+import {
+  ANVIL_KEY,
+  REGISTRY_PATH,
+  deployOrUseExistingCore,
+  extendWarpConfig,
+  getChainId,
+  updateOwner,
+} from './commands/helpers.js';
+import { hyperlaneWarpDeploy, readWarpConfig } from './commands/warp.js';
+
+/// To run: 1) start 2 anvils, 2) yarn run tsx tests/warp.zs-test.ts inside of cli/
+const CHAIN_NAME_1 = 'anvil1';
+const CHAIN_NAME_2 = 'anvil2';
+
+const BURN_ADDRESS = '0x0000000000000000000000000000000000000001';
+const EXAMPLES_PATH = './examples';
+const CORE_CONFIG_PATH = `${EXAMPLES_PATH}/core-config.yaml`;
+const WARP_CONFIG_PATH = `${EXAMPLES_PATH}/warp-route-deployment.yaml`;
+const WARP_CORE_CONFIG_PATH_1 = `${REGISTRY_PATH}/deployments/warp_routes/ETH/anvil1-config.yaml`;
+
+describe('WarpApply e2e tests', async function () {
+  this.timeout(0); // No limit timeout since these tests can take a while
+
+  let chain2Addresses: ChainAddresses = {};
+  before(async () => {
+    await deployOrUseExistingCore(CHAIN_NAME_1, CORE_CONFIG_PATH);
+    chain2Addresses = await deployOrUseExistingCore(
+      CHAIN_NAME_2,
+      CORE_CONFIG_PATH,
+    );
+  });
+
+  beforeEach(async function () {
+    await hyperlaneWarpDeploy(WARP_CONFIG_PATH);
+  });
+
+  it('should burn owner address', async function () {
+    const warpConfigPath = `${EXAMPLES_PATH}/warp-route-deployment-2.yaml`;
+    await updateOwner(
+      BURN_ADDRESS,
+      CHAIN_NAME_1,
+      warpConfigPath,
+      WARP_CORE_CONFIG_PATH_1,
+    );
+    const updatedWarpDeployConfig = await readWarpConfig(
+      CHAIN_NAME_1,
+      WARP_CORE_CONFIG_PATH_1,
+      warpConfigPath,
+    );
+    expect(updatedWarpDeployConfig.anvil1.owner).to.equal(BURN_ADDRESS);
+  });
+
+  it('should not update the same owner', async () => {
+    const warpConfigPath = `${EXAMPLES_PATH}/warp-route-deployment-2.yaml`;
+    await updateOwner(
+      BURN_ADDRESS,
+      CHAIN_NAME_1,
+      warpConfigPath,
+      WARP_CORE_CONFIG_PATH_1,
+    );
+    const { stdout } = await updateOwner(
+      BURN_ADDRESS,
+      CHAIN_NAME_1,
+      warpConfigPath,
+      WARP_CORE_CONFIG_PATH_1,
+    );
+
+    expect(stdout).to.include(
+      'Warp config on anvil1 is the same as target. No updates needed.',
```

Comment:
We should probably abstract error messages + codes into a const file if we're going to start testing them like this. You don't have to do that now

Reply:
Definitely considered this! However, i'm not sure if i will keep this style of testing. still investigating here, but will keep this for now